### DISCUSSION TITLE
7.0.12-RC01

### DISCUSSION THREAD

**Discussion by MikeNeilson** [opendcs/opendcs#454] [Announcements]
## What's Changed
* JDBC bind variables for LoadingAppDAO by @MikeNeilson in https://github.com/opendcs/opendcs/pull/347
* Fix profile handling by @MikeNeilson in https://github.com/opendcs/opendcs/pull/412
* Fix regression in DECODES Message Browser by @MikeNeilson in https://github.com/opendcs/opendcs/pull/414
* Update README.md by @MikeNeilson in https://github.com/opendcs/opendcs/pull/420
* Bugfix/429 configeditor over commiting objects by @MikeNeilson in https://github.com/opendcs/opendcs/pull/431
* Bugfix/screening connection retrieval by @MikeNeilson in https://github.com/opendcs/opendcs/pull/316
* JDBC bind variables for XmitRecordDao by @MikeNeilson in https://github.com/opendcs/opendcs/pull/432
* Some documentation cleanup while reading Computation Processor User Guide by @ktarbet in https://github.com/opendcs/opendcs/pull/450
* JDBC Bind variables in DacqEventDAO improvements by @MikeNeilson in https://github.com/opendcs/opendcs/pull/444
* JDBC Bind variables/organization in comp depends system by @MikeNeilson in https://github.com/opendcs/opendcs/pull/441

**Full Changelog**: https://github.com/opendcs/opendcs/compare/7.0.11...7.0.12-RC01

Known issues:
- There's still some GUI lists that don't handle sorting correctly.
- The routingscheduler is not totally stable but should survive at least 3-4 days and fully crash on failure instead of getting into an effectively deadlocked state.

<hr /><em>This discussion was created from the release <a href='https://github.com/opendcs/opendcs/releases/tag/7.0.12-RC01'>7.0.12-RC01</a>.</em>

**Comment by tracyfraley1** [comment]
We use a script to call a routing spec for an Abstract Web data source. We collect data for a few iridium platforms using a dynamic call to the USGS. With this new version, it calls the rs command. It seems to work but gets hungup. This results in the lockfile not getting deleted, and the call fails the next time. 

WARNING: JOOQ CWMSDB API implemented for CWMS database schema version 18.1.21, connecting to version 23.3.16

I'm not sure if this warning actually is causing the issue. It's just the only output I am seeing. I can run the call at the command line. It hangs up until I kill it. Then it seems to have worked. This did not happen in our older OpenDCS version (6.8.07)

**Reply by MikeNeilson** [reply]
That's not causing the issue, but we've certainly done enough changes that one of them could be.

Let me see if I can yank the configuration out of your database and import it it to recreate it locally.

**Reply by MikeNeilson** [reply]
Oh, follow up question, is it locking up immediately and completely, or is it that it runs for a day or two and you still see logo messages going through but you get no data?

**Reply by tracyfraley1** [reply]
The command is format is 
rs -k lockfile AbstractWebRoutingSpec

This is ran by a script twice an hour to pull in a few gages off the USGS web api. (Iridium gages or else we'd use GOES)

Whether I use the script or just run it directly at the command line, it takes an excessive amount of time to just hang. We noticed the lockfile still existed an hour after the script ran. I tested at the command line and killed it after 10 minutes. It does seem to process the data. Just never disconnects? 

Yesterday I tested with the command line and the script by modifying it to 
timeout 60 rs -k lockfile AbstractWebRoutingSpec

This seemed to work, but I would like it to finish cleanly like it had previously done.

**Reply by MikeNeilson** [reply]
Try it again with the the -k lockfile. It's not, and can't be, a continous routing spec which is where that would come in.

**Reply by tracyfraley1** [reply]
I tried without the lockfile and that worked. Before we started using a continuous running Routing Scheduler to collect our GOES data, we used a rs command ran by the cron. It used a lockfile, and I did not realize the lockfile is only needed if it is continuously running. Thank you for the help.

**Reply by MikeNeilson** [reply]
That works for the GOES data because the feed itself, from the LRGS, is continuous. Whereas for the web sources you get the current state and it's done.

Probably something we should try to make more clear in the documentation.
@zamusacelre FYI, see above chain.
